lib/util/collectionEmulator.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  INTERNAL_PREFIX = '%',
+  USER_PREFIX = '&',
```

Comment:
(nitpicking) Wording: I prefer speaking of public indexes vs "user" indexes (for instance, plugin indexes are private, but can be considered as user indexes if the plugin is a business-specific one)

```suggestion
  PUBLIC_PREFIX = '&',
```